### DISCUSSION TITLE
How to link ogbn-arxiv with titleabs.tsv ?

### DISCUSSION THREAD

**Discussion by brochier** [snap-stanford/ogb#175] [General]
In the description of the ogbn-arxiv dataset, it is written:

> We also provide the mapping from MAG paper IDs into the raw texts of titles and abstracts here.

I opened this file and found 179,717 titles and abstracts with their respective MAG ids. How could I link these text features with the original graph ?

**Comment by weihua916** [comment]
Hi!

The first number represents the MAG ID. For instance https://academic.microsoft.com/paper/200971/
You can use the information in `${ROOT}/ogbn_arxiv/mapping/nodeidx2paperid.csv.gz` to map node index to MAG ID, which can then be used to obtain the raw text using titleabs.tsv

===== titleabs.tsv

200971	ontology as a source for rule generation	This paper discloses the potential of OWL (Web Ontology Language) ontologies for generation of rules. The main purpose of this paper is to identify new types of rules, which may be generated from OWL ontologies. Rules, generated from OWL ontologies, are necessary for the functioning of the Semantic Web Expert System. It is expected that the Semantic Web Expert System (SWES) will be able to process ontologies from the Web with the purpose to supplement or even to develop its knowledge base.

**Reply by brochier** [reply]
That's great, thanks for your quick reply !
